tests/e2e/k8s/util/kind.go

```diff
@@ -0,0 +1,425 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/support/kind"
+)
+
+type ServiceNotFoundError struct{}
+
+func (e ServiceNotFoundError) Error() string {
+	return "service not found"
+}
+
+type ConnectionRefusedError struct{}
+
+func (e ConnectionRefusedError) Error() string {
+	return "connection refused"
+}
+
+type ConnectionResetError struct{}
+
+func (e ConnectionResetError) Error() string {
+	return "connection reset"
+}
+
+// Service represents a kubernetes service.
+type Service struct {
+	// Name is the service name.
+	Name string
+	// Namespace is the service namespace.
+	Namespace string
+	// Port is the service external listening port.
+	Port uint16
+}
+
+// PodAndService represents a kubernetes service and a backing pod.
+type PodAndService struct {
+	Service
+
+	// Image is the container image.
+	Image string
+	// Args are the container command line arguments.
+	Args []string
+	// TargetPort is the internal pod port.
+	TargetPort uint16
```

Comment:
This should be part of the service

Reply:
As you can see, I'm defining a new struct, and not using k8s objects. So I'm not obliged to match the k8s definition of objects.
The `Service` struct is used for consumers of the service.
The only entity which cares about the targetport is the pod (defined by `PodAndService`).

Reply:
But you use it to create a service, you can decide that the port and the targetport use the same value and remove this field.
But if you want in the future (or even now) to split the function CreatePodAndService to  CreatePod and  CreateService you need this field in the CreateService function.

Reply:
removed targetport